wasm/jit/jit_arm64.go

```diff
@@ -554,8 +624,229 @@ func (c *arm64Compiler) compileBrTable(o *wazeroir.OperationBrTable) error {
 	return fmt.Errorf("TODO: unsupported on arm64")
 }
 
+// compileCall implements compiler.compileCall for the arm64 architecture.
 func (c *arm64Compiler) compileCall(o *wazeroir.OperationCall) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	target := c.f.ModuleInstance.Functions[o.FunctionIndex]
+
+	if err := c.callFunction(target.Address, target.FunctionType.Type); err != nil {
+		return err
+	}
+	return nil
+}
+
+// compileCall implements compiler.compileCall and compiler.compileCallIndirect (TODO) for the arm64 architecture.
+func (c *arm64Compiler) callFunction(addr wasm.FunctionAddress, functype *wasm.FunctionType) error {
+	// TODO: the following code can be generalized for CallIndirect.
+
+	// Release all the registers as our calling convention requires the caller-save.
+	if err := c.releaseAllRegistersToStack(); err != nil {
+		return err
+	}
+
+	// Obtain the temporary registers to be used in the followings.
```

Comment:
somewhere above here, we claimed 3 registers and their usage was visually close (a couple lines below all were declared. That's not true here, so we may need to describe them.

In fact, I prefer special-casing so you can declare inline. Ex define

`takeFreeRegister` `takeTwoFreeRegisters` (if ever there's 2) and `takeThreeFreeRegisters`, and that way you can use the results immediately and prevent the indirection of what they are used for

ex.

```
purposeX, purposeY, purposeZ, found := c.locationStack.takeThreeFreeRegisters(generalPurposeRegisterTypeInt)
```

Reply:
in anyway, we end up literally having `temporaryRegisterX` `temporaryRegisterY` and they are no different from using `tmpRegs[0]` and `tmpReg[1]` and that's why I intentionally used `tmpRegs[0, 1, ...]` and  only alias the ones with purpose

Reply:
but I understood the concern, let me refine